### PR TITLE
Website update in preparation for the 2.5.0 release.

### PR DESCRIPTION
**Pull request by jrfnl**
**To be merged on releasing 2.5.0**

Preview available: http://jrfnl.github.io/TGM-Plugin-Activation/

Highlights:
* Jekyll
	- Converted the website to use jekyll instead of plain HTML to make working with several pages easier to manage (common headers, navigation, footers etc).
	- Removed syntax highlighting js and css files in favour of using Jekyll with Pygments.
	- I've included a number of example files in the `_draft` folder which contain quite detailed instructions on how to use these in the context of Jekyll.

* Jekyll/GitHub integration
	- Changed download links to always point to the last release zip/tar package instead of to the master branch.
	- Added a 'release archive' with links to the five previous releases.
	- Added a changelog subsection on the download page which will show the release notes of the last release.
	- Added contributors list via a js AJAX call to the GitHub API.
	- All of the above have fall-backs for when this information would not be available. Some of the above use js to inject the information as the TGMPA releases have been added at a later date and the data needs to be sorted and manipulated, which turned out to be impossible to do using Jekyll.

* TGMPA organization
	- Changed references to the old repo location to the new one.
	- Added the new logo to the page.
	- Created a favicon based on the new logo.

* 2.5.0 release
	- Changed references to 'pre-packaged' plugins to 'bundled'.
	- Updated the example.php code to the current version in `develop`, renamed the API section to `Configuration` and moved it to a separate page.
	- Updated the plugin settings and configuration option sections and changed the layout to a much more readable table format.
	- Added contributing section under authors.
	- Added roadmap page containing info on the v3.0 roadmap.
	- Added FAQ page with sub-pages.
	- Updated two of the four screenshots.

* Other:
	- Renamed 'Dependencies' section to 'Requirements'.
	- Added current version number to the page header.
	- Added last update date of the site in the page footer.
	- Added Twitter follow and Tweet buttons.
	- Added blog section (currently unused - an example of what it would look like when used can be seen in my fork http://jrfnl.github.io/TGM-Plugin-Activation/ ).
	- Added 404 page.
	- Lots of small css tweaks.
	- Added marked.js library to convert release notes from markdown to html.
	- Added tgmpa.js file which handles the redirects and the injecting of GH releases info. The minified version will be used on the live site. The non-minified version will automatically be used when in a test environment.
	- Added robots.txt.
	- Added Atom feed for blog.
	- Added sitemap.xml + xml-sitemap.xsl - this does *not* use the Jekyll sitemaps plugin as it's not flexible enough. Sitemap code inspired by http://davidensinger.com/2013/11/building-a-better-sitemap-xml-with-jekyll/
	- Added appropriate redirects for links from the old site using js (as you can't redirect #hashes through .htaccess).
	- Added separate contributing instructions for the gh-pages branch.
	- Added travis-ci build testing for jekyll.

__*Once this branch has been merged, the FAQ and removal instructions wiki pages should be removed.*__

### CODE REVIEW

File: _faqs/03-uninstalling-tgmpa.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	 */
+	function prefix_register_plugins() {
+
+		/**
+		 * Array of plugin arrays. Required keys are name and slug.
+		 * If the source is NOT from the .org repo, then source is also required.
+		 */
+		$plugins = array(
+
+			// Wordpress SEO
```
Comment (grappler) [inline _faqs/03-uninstalling-tgmpa.md]: WordPress is spelled incorrectly here :tongue:

File: 04-Configuration.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+anchor:    configuration
+permalink: /configuration/
+---
+
+### Configuring TGMPA for use with your theme/plugin
+
+
+The **{{site.tgmpa.name}}** library has an easy to use API to reference plugins and set config options. Consider the code below (taken from `example.php`).
+
+**_N.B.: this code may be slightly out of date - the download package will always contain the most up-to-date version._**
```
Comment (grappler) [inline 04-Configuration.md]: It should also be mentioned that the code is not complete as the strings are missing. To keep the code snippet shorter I would only have one example in the plugin array.

File: 03-Installation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Installing the **{{site.tgmpa.name}}** library is easy. Follow the instructions below:
+
+1. [Download] the latest release.
+2. Drop the class file somewhere in your theme or plugin hierarchy.
+3. Add a `require_once` call within `functions.php` (or other theme or plugin file) referencing the class file.
+4. Create a function, hooked to `tgmpa_register`, that registers the plugin and configurations.
+
+
+For steps 3 and 4, it is **_strongly recommended_** you view, copy and paste the contents of the `example.php` file which is included in the release package and amend to suit. The `example.php` file is a model for how you should include the class in your theme.
```
Comment (grappler) [inline 03-Installation.md]: It would be useful if we could link example.php to GitHub.

File: _faqs/03-uninstalling-tgmpa.md (hunk 1 above)
Comment (GaryJones) [inline _faqs/03-uninstalling-tgmpa.md]: That's it @jrfnl, you're fired!

Comment (jrfnl) [inline _faqs/03-uninstalling-tgmpa.md]: @GaryJones Actually, you are: https://github.com/TGMPA/TGM-Plugin-Activation/wiki/Removing-TGM-Plugin-Activation-From-Your-Theme-Or-Plugin

Comment (GaryJones) [inline _faqs/03-uninstalling-tgmpa.md]: I (blindly) copied that from someone who emailed in with a question and attached their code, so... :dog:

File: 04-Configuration.md (hunk 2 above)
Comment (jrfnl) [inline 04-Configuration.md]: Added a big notice at the top of the page ;-)

File: 03-Installation.md (hunk 3 above)
Comment (jrfnl) [inline 03-Installation.md]: @grappler I could, but am a bit wary - I'd prefer if people use the file included in their download as that's the one belonging to the version they have.

**Comment by jrfnl** [comment]
PR updated - most significant change: added "Since" column to the tables on the configuration page with version number indications showing which plugin parameter and which configuration option has been available since which TGMPA version.

**Comment by GaryJones** [comment]
Good to merge when 2.5.0 is released.

**Pull request merged**